publication-commons/src/main/java/no/unit/nva/publication/permissions/file/grant/EveryoneGrantStrategy.java

```diff
@@ -16,11 +16,14 @@ public EveryoneGrantStrategy(File file,
 
     @Override
     public boolean allowsAction(FileOperation permission) {
-        if (FileOperation.READ_METADATA == permission) {
-            //TODO make sure embargo and other restrictions are checked
-            // and download is allowed
-            return file instanceof OpenFile;
+        if (file instanceof OpenFile openFile) {
+            return switch (permission) {
+                case READ_METADATA -> true;
+                case DOWNLOAD -> openFile.fileDoesNotHaveActiveEmbargo();
+                default -> false;
+            };
         }
+
```

Comment:
What about using new **when** condition in switch clause:

```
return switch (file) {
            case OpenFile ignored when permission.equals(FileOperation.READ_METADATA) -> true;
            case OpenFile openFile when permission.equals(FileOperation.DOWNLOAD) -> openFile.fileDoesNotHaveActiveEmbargo();
            default -> false;
        };
```

Reply:
pretty cool! but I think in this case it would not grow great as we adapt more complexity and it may become hard to see if we have exhausted all needed possibilities